test/jdk/java/lang/management/ThreadMXBean/VirtualThreads.java

```diff
@@ -63,32 +63,56 @@ public class VirtualThreads {
      */
     @Test
     public void testGetAllThreadIds() throws Exception {
-        runInVirtualThread(() -> {
-            long currentTid = Thread.currentThread().getId();
+        Thread vthread = Thread.startVirtualThread(LockSupport::park);
+        try {
             long[] tids = ManagementFactory.getThreadMXBean().getAllThreadIds();
-            boolean found = Arrays.stream(tids).anyMatch(tid -> tid == currentTid);
-            assertFalse(found);
-        });
+
+            // current thread should be included
+            long currentTid = Thread.currentThread().threadId();
+            assertTrue(Arrays.stream(tids).anyMatch(tid -> tid == currentTid));
+
+            // virtual thread should not be included
+            long vtid = vthread.threadId();
+            assertFalse(Arrays.stream(tids).anyMatch(tid -> tid == vtid));
+        } finally {
+            LockSupport.unpark(vthread);
+        }
     }
 
     /**
-     * Test that ThreadMXBean::getThreadInfo returns null for a virual thread.
+     * Test that ThreadMXBean.getThreadInfo(long) returns null for a virtual thread.
      */
     @Test
     public void testGetThreadInfo1() throws Exception {
+        Thread vthread = Thread.startVirtualThread(LockSupport::park);
+        try {
+            long tid = vthread.threadId();
+            ThreadInfo info = ManagementFactory.getThreadMXBean().getThreadInfo(tid);
+            assertTrue(info == null);
+        } finally {
+            LockSupport.unpark(vthread);
+        }
+    }
+
+    /**
+     * Test that ThreadMXBean.getThreadInfo(long) returns null when invoked by a virtual
+     * thread with its own thread id.
+     */
+    @Test
+    public void testGetThreadInfo2() throws Exception {
         runInVirtualThread(() -> {
-            long tid = Thread.currentThread().getId();
+            long tid = Thread.currentThread().threadId();
             ThreadInfo info = ManagementFactory.getThreadMXBean().getThreadInfo(tid);
             assertTrue(info == null);
         });
     }
 
     /**
-     * Test that ThreadMXBean::getThreadInfo on a carrier thread. The stack
-     * frames of the virtual thread should not be returned.
+     * Test that ThreadMXBean.getThreadInfo(long) with the thread id of a carrier thread.
```

Comment:
Pre-existing typo: this comment sentence looks truncated?  Or maybe just remove "that" and it's fine: 
"Test ThreadMXBean.getThreadInfo(long) with.."